embedded/store/immustore.go

```diff
@@ -2571,13 +2571,17 @@ func (s *ImmuStore) readValueAt(b []byte, off int64, hvalue [sha256.Size]byte, s
 		val, err := s.vLogCache.Get(off)
 		if err == nil {
 			// the requested value was found in the value cache
-			copy(b, val.([]byte))
+			bval := val.([]byte)
+			n := len(bval)
 
-			if !skipIntegrityCheck && hvalue != sha256.Sum256(b) {
-				return len(b), ErrCorruptedData
+			copy(b, bval)
+
+			// length is compared to cover the corner case where empty value is stored in the cache
+			if !skipIntegrityCheck && (len(b) != n || hvalue != sha256.Sum256(b[:n])) {
+				return n, ErrCorruptedData
 			}
 
-			return len(b), nil
+			return n, nil
 		} else {
 			if !errors.Is(err, cache.ErrKeyNotFound) {
 				return 0, err
```

Comment:
return n, err?

Reply:
err is nil and so, having it explicit simplifies reading the code

Reply:
it is not really consistent with the changes being made

Reply:
don't see like that..  

I figured out that truncated transaction in the replicas should receive the value length as it's receiving the digest, otherwise it becomes inconsistent with the data hold in master after truncation, making impossible to differentiate empty value with non-empty ones during integrity checks